tests/utility/make_unique.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright (c) 2014 Artem Sokolov <derikon@yandex.ru>
+Copyright (c) 2011-2014 Other contributors as noted in the AUTHORS file.
+
+This file is part of Kora.
+
+Kora is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Kora is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <gtest/gtest.h>
+
+#include "kora/utility/make_unique.hpp"
+#include "kora/utility/sfinae.hpp"
+
+namespace {
+
+struct a_t {
+    a_t() : x(0), y(0) { }
+    a_t(int x_) : x(x_), y(0) { }
```

Comment:
Do we really need all three structs then? What does it test?
If it tests that make_unique correctly forwards arguments to the object's constructor, then maybe it should be a separate testcase?